be2-scala/src/main/scala/ch/epfl/pop/decentralized/ConnectionMediator.scala

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.pop.decentralized
+
+import akka.actor.{Actor, ActorLogging, ActorRef, ActorSystem, Props}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.ws.WebSocketRequest
+import akka.pattern.AskableActorRef
+
+import ch.epfl.pop.model.network.method.{Heartbeat, ParamsWithMap}
+import ch.epfl.pop.model.network.{JsonRpcRequest, MethodType}
+import ch.epfl.pop.pubsub.ClientActor.ClientAnswer
+import ch.epfl.pop.pubsub.graph.validators.RpcValidator
+import ch.epfl.pop.pubsub.{AskPatternConstants, MessageRegistry, PublishSubscribe}
+final case class ConnectionMediator(
+    monitorRef: ActorRef,
+    mediatorRef: ActorRef,
+    dbActorRef: AskableActorRef,
+    messageRegistry: MessageRegistry
+) extends Actor with ActorLogging with AskPatternConstants {
+  implicit val system: ActorSystem = ActorSystem()
+
+  // List of servers connected
+  private var serverSet: Set[ActorRef] = Set()
+
+  override def receive: Receive = {
+
+    // Connect to some servers
+    case ConnectionMediator.ConnectTo(urlList) =>
+      urlList.map(url =>
+        Http().singleWebSocketRequest(
+          WebSocketRequest(url),
+          PublishSubscribe.buildGraph(
+            mediatorRef,
+            dbActorRef,
+            messageRegistry
+          )
+        )
+      )
+
+    case ConnectionMediator.NewServerConnected(serverRef) =>
+      log.info("Received new server")
+      if (serverSet.isEmpty)
+        monitorRef ! Monitor.AtLeastOneServerConnected
+
+      serverSet += serverRef
```

Comment:
I don't know much about the server-to-server communication protocol so this may not make much sense in a broader perspective, but wouldn't that be helpful once a new server connects to this server to send it the list of servers we already are connected to (basically adding here something like `serverRef ! urlList` where `urlList` would be an up to date list of the urls of the servers this server is connected to)?

Reply:
This could be a good idea but as far as I can tell there is nothing in the protocol to supports that properly (missing the jsons). Though it could definitely be something that we add later on !